libjsqrc/ethereumjs/test/abi.parsers.js

```diff
@@ -1,37 +1,538 @@
+            
+        });
+
+        it('should parse input bool', function() {
+            
+            // given
+            var d = clone(description);
+
+            d[0].inputs = [
+                { type: 'bool' }
+            ];
+
+            // when
+            var parser = abi.inputParser(d);
+
+            // then
+            assert.equal(parser.test(true), "0000000000000000000000000000000000000000000000000000000000000001");
+            assert.equal(parser.test(false), "0000000000000000000000000000000000000000000000000000000000000000");
+
+        });
+
+        it('should parse input hash', function() {
+
+            // given
+            var d = clone(description);
+
+            d[0].inputs = [
+                { type: "hash" }
+            ];
+            
+            // when
+            var parser = abi.inputParser(d);
+
+            // then
+            assert.equal(parser.test("0x407d73d8a49eeb85d32cf465507dd71d507100c1"), "000000000000000000000000407d73d8a49eeb85d32cf465507dd71d507100c1");
+
+        }); 
+
+        it('should parse input hash256', function() {
+
+            // given
+            var d = clone(description);
+
+            d[0].inputs = [
+                { type: "hash256" }
+            ];
+
+            // when
+            var parser = abi.inputParser(d);
+
+            // then
+            assert.equal(parser.test("0x407d73d8a49eeb85d32cf465507dd71d507100c1"), "000000000000000000000000407d73d8a49eeb85d32cf465507dd71d507100c1");
+
+        });
+
+
+        it('should parse input hash160', function() {
+            // given
+            var d = clone(description);
+
+            d[0].inputs = [
+                { type: "hash160" }
+            ];
+
+            // when
+            var parser = abi.inputParser(d);
+            
+            // then
+            assert.equal(parser.test("0x407d73d8a49eeb85d32cf465507dd71d507100c1"), "000000000000000000000000407d73d8a49eeb85d32cf465507dd71d507100c1");
+        });
+
+        it('should parse input address', function () {
+
+            // given
+            var d = clone(description);
+
+            d[0].inputs = [
+                { type: "address" }
+            ];
+            
+            // when
+            var parser = abi.inputParser(d)
+            
+            // then
+            assert.equal(parser.test("0x407d73d8a49eeb85d32cf465507dd71d507100c1"), "000000000000000000000000407d73d8a49eeb85d32cf465507dd71d507100c1");
+
+        });
+
+        it('should parse input string', function () {
+            
+            // given
+            var d = clone(description);
+
+            d[0].inputs = [
+                { type: "string" }
+            ];
+
+            // when
+            var parser = abi.inputParser(d);
+
+            // then
+            assert.equal(parser.test('hello'), "68656c6c6f000000000000000000000000000000000000000000000000000000");
```

Comment:
It should be "056865..." (same amount of zeros): Since string has variable length, you have to prepend its length.
Note that this type is not supported by solidity yet, only the fixed-length strings up to 32 bytes.